graphbolt/src/cuda/utils.h

```diff
@@ -0,0 +1,36 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *
+ * @file utils.h
+ * @brief CUDA utilities.
+ */
+
+#ifndef GRAPHBOLT_CUDA_UTILS_H_
+#define GRAPHBOLT_CUDA_UTILS_H_
+
+namespace graphbolt {
+namespace cuda {
+
+// The cache line size of GPU
+#define GPU_CACHE_LINE_SIZE 128
+// The max number of threads per block
+#define CUDA_MAX_NUM_THREADS 1024
+
+/** @brief Calculate the number of threads needed given the dimension length.
+ *
+ * It finds the biggest number that is smaller than min(dim, max_num_nthreads)
```

Comment:
It finds the largest power of two that is less than or equal to the minimum of dim and max_num_threads.